src/api/result_proxy_api_client.rs

```diff
@@ -0,0 +1,359 @@
+use reqwest::blocking::Client;
+use serde::{Deserialize, Serialize};
+
+const EMPTY_HEX_STRING_32: &str =
+    "0x0000000000000000000000000000000000000000000000000000000000000000";
+const EMPTY_WEB3_SIG: &str = "0x000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000";
+
+/// Represents a computation result that can be uploaded to IPFS via the iExec result proxy.
+///
+/// This struct encapsulates all the necessary information about a completed computation task
+/// that needs to be stored permanently on IPFS. It includes task identification, metadata,
+/// the actual result data, and cryptographic proofs of computation integrity.
+///
+/// The struct is designed to be serialized to JSON for transmission to the result proxy API,
+/// with field names automatically converted to camelCase to match the expected API format.
+///
+/// # Fields
+///
+/// * `chain_task_id` - The unique identifier of the task on the blockchain
```

Comment:
In Rust standard library, fields documentation seems to be made directly on each field of the struct:
Here is an example for the `Option` and `Result` structs:
https://doc.rust-lang.org/src/core/option.rs.html#572
https://doc.rust-lang.org/src/core/result.rs.html#528